internal/util/context.go

```diff
@@ -19,10 +19,10 @@ func ContextWithLogger(ctx context.Context, logger *logrus.Entry) context.Contex
 
 // LoggerFromContext returns the logger from the context.
 func LoggerFromContext(ctx context.Context) *logrus.Entry {
-	logger := ctx.Value(loggerKey).(*logrus.Entry)
+	logger := ctx.Value(loggerKey)
 	if logger == nil {
 		logger = logrus.NewEntry(logrus.New())
 	}
 
-	return logger
+	return logger.(*logrus.Entry)
```

Comment:
I think this is still wrong and could still panic (what if we don't get *logrus.Entry type, but something else?).

Do this:

```
v, ok := logger.(*logrus.Entry)
if !ok {
	logrus.Warnf("received wrong type of logger (%T)", logger)
    // initialize default logger as above
}
```

Reply:
I added check for nil and for wrong type. I also created tests for this.

I kept check for nil because when nil, the error message was ""received wrong type of logger (*logrus.Entry)", which is what we actually want and might be confusing.

Reply:
> the error message was "received wrong type of logger (*logrus.Entry)", which is what we actually want and might be confusing.

these error messages help us as developers to fix it.
but it is meant to prevent the process from crashing so users can continue to use it if we accidentally create a bug by passing loggerless context.